apple/Sources/ValhallaObjc/ValhallaWrapper.mm

```diff
@@ -4,25 +4,31 @@
 
 @implementation ValhallaWrapper
 
-- (instancetype)init
+- (instancetype)initWithConfigPath:(NSString*)config_path
 {
     self = [super init];
+    std::string path = std::string([config_path UTF8String]);
+    _actor = create_valhalla_actor(path.c_str());
     return self;
 }
 
-- (NSString*)route:(NSString*)request configPath:(NSString*)config_path
+- (NSString*)route:(NSString*)request
 {
-    // Get the config path
-    std::string path = std::string([config_path UTF8String]);
-    std::string config_file(path);
-    
-    // Convert the NSString to std::string
-    std::string req = std::string([request UTF8String]);
-    
-    // Generate the valhalla response
-    std::string res = route(req.c_str(), path.c_str());
-    
-    return [NSString stringWithUTF8String:res.c_str()];
+    @synchronized(self) {
```

Comment:
This reminds me why Rust exists 😅 Good call. I assume Valhalla actors are not thread safe?

Reply:
Yes, unfortunately without this the app crashes when multiple concurrent route-calls use the same Valhalla-instance.

Reply:
Makes sense. Most C++ isn't; just wasn't sure if you'd tested explicitly. Thanks!